pkg/db/src/implementations/version-1.2/books.ts

```diff
@@ -37,10 +37,29 @@ class Books implements BooksInterface {
 	}
 
 	async get(isbns: string[]): Promise<(BookEntry | undefined)[]> {
-		const rawBooks = await this.#db._pouch.allDocs<BookEntry>({ keys: isbns.map((isbn) => `books/${isbn}`), include_docs: true });
+		const rawBooks = await this.#db._pouch
+			.allDocs<BookEntry>({ keys: isbns.map((isbn) => `books/${isbn}`), include_docs: true })
+			.then((docs) => unwrapDocs(docs));
 		// The rows are returned in the same order as the supplied keys array.
 		// The row for a nonexistent document will just contain an "error" property with the value "not_found".
-		return unwrapDocs(rawBooks);
+
+		const combinedResults = wrapIter(isbns).zip(rawBooks);
+
+		const existingBooks: BookEntry[] = [];
+		// temp fix for isbn
+		const fetchIsbns = combinedResults.map(([isbn, book]) => {
+			if (book !== undefined) {
+				existingBooks.push(book);
+				return;
+			}
+
+			return isbn;
+		});
```

Comment:
1. With respect to the comments below, you'll see that `existingBooks` is unnecessary
2. There's a bug: when you do `.map(() => { ... return isbn }` you're returning all isbns (both found and not found)
3. I would prefer it if we used the more functional approach: to `wrapIter().zip()` you can also chain `.filter` (filtering out the books that are known, leaving you with only ones to fetch) and `.map` to return only the isbn

Reply:
_Note: the `wrapIter` works with generators, so it doesn't matter how many "iterations" you chain together as they're iterated over (and each step applied) only when the iterable is consumed_